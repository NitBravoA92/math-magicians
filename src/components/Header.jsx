import Logo from './Logo';
import Navigation from './Navigation';
import '../styles/Header.css';

const Header = () => (
  <header>
    <Logo />
    <Navigation />
  </header>
);

export default Header;
