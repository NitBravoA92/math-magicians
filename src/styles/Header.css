header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  background-color: whitesmoke;
  padding: 0.5rem 1.2rem 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1.5rem;
  box-shadow: rgba(0, 0, 0, 0.1) 0 4px 6px -1px, rgba(0, 0, 0, 0.06) 0 2px 4px -1px;
}

.logo-container {
  display: flex;
  justify-content: center;
  align-items: flex-end;
  gap: 1rem;
}

.App-logo {
  width: 60px;
}

.logo-title {
  font-size: 1.8em;
  font-weight: 900;
  color: #121111;
}

#menu {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.5rem;
}

#menu .nav-item a {
  font-size: 1em;
  color: #121111;
  padding: 0.8rem 0.6rem;
  position: relative;
}

#menu .nav-item a.active {
  color: #f50057;
}

#menu .nav-item a:hover {
  color: #f50057;
}

#menu .nav-item:nth-child(2) a::before,
#menu .nav-item:nth-child(2) a::after {
  content: "";
  width: 2px;
  height: 95%;
  background-color: #f50057;
  position: absolute;
  top: 0;
}

#menu .nav-item:nth-child(2) a::before {
  left: 0;
}

#menu .nav-item:nth-child(2) a::after {
  right: 0;
}

@media (min-width: 768px) {
  header {
    flex-direction: row;
    justify-content: space-between;
  }

  .logo-title {
    font-size: 2.5em;
  }

  #menu {
    height: 40px;
    align-items: flex-end;
  }
}
