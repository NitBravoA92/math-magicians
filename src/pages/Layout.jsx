import PropTypes from 'prop-types';
import Header from '../components/Header';

const Layout = ({ children }) => (
  <>
    <Header />
    <main>
      { children }
    </main>
  </>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
