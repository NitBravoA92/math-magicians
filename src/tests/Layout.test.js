import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../pages/Layout';
import Header from '../components/Header';

describe('The Layout component', () => {
  test('renders correctly into the DOM', () => {
    const component = renderer
      .create(
        <Layout>
          <Header />
          <main>
            <section id="home">
              <h1 className="page-title">Welcome to our page!</h1>
            </section>
          </main>
        </Layout>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
