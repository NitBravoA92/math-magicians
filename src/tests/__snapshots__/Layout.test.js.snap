// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Layout component renders correctly into the DOM 1`] = `
Array [
  <header>
    <a
      className="logo-container"
      href="/"
    >
      <img
        alt="logo"
        className="App-logo"
        src="mathLogo.gif"
      />
      <span
        className="logo-title"
      >
        Math Magicians
      </span>
    </a>
    <nav>
      <ul
        id="menu"
      >
        <li
          className="nav-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <header>
      <a
        className="logo-container"
        href="/"
      >
        <img
          alt="logo"
          className="App-logo"
          src="mathLogo.gif"
        />
        <span
          className="logo-title"
        >
          Math Magicians
        </span>
      </a>
      <nav>
        <ul
          id="menu"
        >
          <li
            className="nav-item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main>
      <section
        id="home"
      >
        <h1
          className="page-title"
        >
          Welcome to our page!
        </h1>
      </section>
    </main>
  </main>,
]
`;
